src/lib/substrate/context/provider.tsx

```diff
@@ -0,0 +1,242 @@
+import { createInterBtcApi } from '@interlay/interbtc-api';
+import { ApiPromise } from '@polkadot/api';
+import { web3Accounts, web3Enable, web3FromAddress } from '@polkadot/extension-dapp';
+import { KeyringPair } from '@polkadot/keyring/types';
+import { TypeRegistry } from '@polkadot/types/create';
+import jsonrpc from '@polkadot/types/interfaces/jsonrpc';
+import { keyring } from '@polkadot/ui-keyring';
+import { isTestChain } from '@polkadot/util';
+import * as React from 'react';
+import { useLocalStorage } from 'react-use';
+
+import { APP_NAME } from '@/config/relay-chains';
+import { SELECTED_ACCOUNT_LOCAL_STORAGE_KEY } from '@/config/wallets';
+
+import * as constants from '../../../constants';
+import { substrateReducer } from './reducer';
+import {
+  _KeyringPair,
+  ActionType,
+  ApiStatus,
+  Dispatch,
+  KeyringStatus,
+  State,
+  SubstrateProviderProps,
+  SubstrateStateContextInterface
+} from './types';
+
+const UNABLE_TO_INITIALISE_OPTIONS_MORE_THAN_ONCE = 'Unable to initialise options more than once';
+
+// /
+// Initial state for `React.useReducer`
+const initialState = {
+  // These are the states
+  socket: undefined,
+  jsonrpc,
+  keyring: undefined,
+  keyringStatus: KeyringStatus.Idle,
+  api: undefined,
+  apiError: undefined,
+  apiStatus: ApiStatus.Idle,
+  selectedAccount: undefined,
+  accounts: [],
+  extensions: []
+};
+
+const registry = new TypeRegistry();
+
+// /
+// Connecting to the Substrate node
+const connect = async (state: State, dispatch: Dispatch) => {
+  try {
+    const { socket } = state;
+
+    dispatch({ type: ActionType.ConnectInit });
+
+    console.log(`Connected socket: ${socket}`);
+
+    window.bridge = await createInterBtcApi(constants.PARACHAIN_URL, constants.BITCOIN_NETWORK);
+    const _api = window.bridge.api;
+
+    dispatch({
+      type: ActionType.Connect,
+      payload: _api
+    });
+    dispatch({ type: ActionType.ConnectSuccess });
+
+    loadAccounts(_api, dispatch);
+
+    // Set listeners for disconnection and reconnection event.
+    _api.on('connected', () => {
```

Comment:
Nit: could you add spacing between functions execution?

Reply:
Done!